### PR TITLE
feat: VFolder trash bin

### PR DESCRIPTION
**Pull request by fregataa**
related issue: #767 

- This pr does not fully cover the related issue because this does not include auto-clearing trash bin.

~~1. impl **trash bin** in vfs storage~~
1. Now, vfolder `delete` API only updates the status column of vfolder table at manager, and `purge` API deletes the real data.
2. add vfolder statuses such as `DELETE_ONGOING` or `PURGE_ONGOING` etc.
3. add vfolder status history column
~~3. impl storage trash bin checker in manager dispatcher.~~
~~`storage_check_interval_days` in manager's shared_config set the interval of trash bin checker.~~

Q: How should we determine duplicate new created vfolder name if we do not delete actual data in DB?
`purge_vfolder` remove the db row and remove the actual vfolder.
The name of `DELETED` vfolders can not be used for new created vfolders but the name of `PURGED` vfolders are available.
-> Decided to remove `PURGED` vfolder row.

### CODE REVIEW

**Review by kyujin-cho — CHANGES_REQUESTED** [review]
We need retention period feature for trash bin. Manager creates a timer for a function which looks for vFolders staying more than designated period at the trash bin and purging them automatically.

**Comment by fregataa** [comment]
TODO
1. Should we include the trash-bin in vfolder metrics?
2. Should users have personal/group-owned trash bin? Because host-unique trash bin makes it hard to determine whose *trash* it is

File: changes/835.feature.md [hunk 1]
```diff
@@ -0,0 +1 @@
+impl minimal vfolder trash bin.
```
Comment (achimnol) [inline changes/835.feature.md]: Please use a full sentence and highlight that this feature provides vfolder-level recovery, not recovery of individual files.
```suggestion
Implement a folder-level trash bin to allow recovery of accidentally deleted vfolders
```

**Review by achimnol — CHANGES_REQUESTED** [review]
* Let's migrate the configuration option to the etcd, as a globally shared config.
* Could you describe what's the context of adding explicit `DoCheckStorageEvent` with logging of the number of vfolders in the trash bin?

**Comment by fregataa** [comment]
> * Could you describe what's the context of adding explicit `DoCheckStorageEvent` with logging of the number of vfolders in the trash bin?

I added `DoCheckStorageEvent` to check trash bin regularly to empty it or log the status of it.

File: src/ai/backend/manager/defs.py [hunk 2]
```diff
@@ -72,6 +72,7 @@ class LockID(enum.IntEnum):
     LOCKID_SCALE_TIMER = 193
     LOCKID_LOG_CLEANUP_TIMER = 195
     LOCKID_IDLE_CHECK_TIMER = 196
+    LOCKID_STORAGE_TIMER = 197
```
Comment (kyujin-cho) [inline src/ai/backend/manager/defs.py]: Looks like this LockID is not used anymore.

File: src/ai/backend/manager/models/vfolder.py [hunk 3]
```diff
@@ -135,8 +140,12 @@ class VFolderOperationStatus(str, enum.Enum):
     READY = "ready"
     PERFORMING = "performing"
     CLONING = "cloning"
-    DELETING = "deleting"
     MOUNTED = "mounted"
+    ERROR = "error"
+    DELETE_ONGOING = "delete-ongoing"  # vfolder is being deleted
+    DELETE_COMPLETE = "deleted-complete"  # vfolder is deleted
+    PURGE_ONGOING = "purge-ongoing"  # vfolder is being removed permanently
+    # PURGE_COMPLETE = "purged-complete"  # vfolder is removed permanently
```
Comment (kyujin-cho) [inline src/ai/backend/manager/models/vfolder.py]: Please remove commented out, unused snippet from the file.

File: src/ai/backend/manager/api/vfolder.py [hunk 4]
```diff
@@ -138,19 +140,17 @@ async def ensure_vfolder_status(
...
                 VFolderOperationStatus.READY,
                 VFolderOperationStatus.PERFORMING,
                 VFolderOperationStatus.CLONING,
-                VFolderOperationStatus.DELETING,
+                VFolderOperationStatus.DELETE_COMPLETE,
                 VFolderOperationStatus.MOUNTED,
             }
         case VFolderAccessStatus.UPDATABLE:
-            # if UPDATABLE access status is requested, READY and MOUNTED operation statuses are accepted.
+            # if UPDATABLE access status is requested, only READY operation status is accepted.
```
Comment (kyujin-cho) [inline src/ai/backend/manager/api/vfolder.py]: Looks like this comment contradicts with actual implementation.

**Review by kyujin-cho — CHANGES_REQUESTED** [review]

**Pull request merged**